File: lollys-lotto-rust-sdk/src/instructions/admin/close_lolly_burn_state.rs
use crate::instructions::*;

pub fn close_lolly_burn_state(
    lolly_burn_state: &Pubkey,
    event_emitter: &Pubkey,
    authority: &Pubkey,
) -> Instruction {
    let data = lollys_lotto::instruction::CloseLollyBurnState.data();
    let accounts = lollys_lotto::accounts::CloseLollyBurnState {
        authority: *authority,
        lolly_burn_state: *lolly_burn_state,
        event_emitter: *event_emitter,
        system_program: system_program::ID,
    }
    .to_account_metas(None);
    Instruction {
        program_id: lollys_lotto::ID,
        accounts,
        data,
    }
}

File: lollys-lotto-ts-sdk/src/codegen/instructions/closeLollyBurnState.ts
import { TransactionInstruction, PublicKey, AccountMeta } from "@solana/web3.js" // eslint-disable-line @typescript-eslint/no-unused-vars
import BN from "bn.js" // eslint-disable-line @typescript-eslint/no-unused-vars
import * as borsh from "@coral-xyz/borsh" // eslint-disable-line @typescript-eslint/no-unused-vars
import * as types from "../types" // eslint-disable-line @typescript-eslint/no-unused-vars
import { PROGRAM_ID } from "../programId"

export interface CloseLollyBurnStateAccounts {
  authority: PublicKey
  lollyBurnState: PublicKey
  eventEmitter: PublicKey
  systemProgram: PublicKey
}

export function closeLollyBurnState(
  accounts: CloseLollyBurnStateAccounts,
  programId: PublicKey = PROGRAM_ID
) {
  const keys: Array<AccountMeta> = [
    { pubkey: accounts.authority, isSigner: true, isWritable: true },
    { pubkey: accounts.lollyBurnState, isSigner: false, isWritable: true },
    { pubkey: accounts.eventEmitter, isSigner: false, isWritable: true },
    { pubkey: accounts.systemProgram, isSigner: false, isWritable: false },
  ]
  const identifier = Buffer.from([81, 45, 206, 37, 231, 154, 202, 61])
  const data = identifier
  const ix = new TransactionInstruction({ keys, programId, data })
  return ix
}
